Fix building for "Windows ia32" on main

It seems building for Windows ia32 broke on main.

I start seeing errors on the merge commit from https://github.com/realm/realm-js/pull/5650: https://github.com/realm/realm-js/actions/runs/4541450640/jobs/8003632690

```
 Checking File Globs
  Checking Build System
  Generating D:/a/realm-js/realm-js/packages/bindgen/generated/spec.schema.json
  Generating D:/a/realm-js/realm-js/packages/realm/generated/ts/native.d.mts, D:/a/realm-js/realm-js/packages/realm/generated/ts/core.ts
  D:\a\realm-js\realm-js\node_modules\@esbuild-kit\core-utils\node_modules\esbuild\lib\main.js:1885
          throw new Error(`The package "${pkg}" could not be found, and is needed by esbuild.

CUSTOMBUILD : error : The package "@esbuild/win32-x64" could not be found, and is needed by esbuild. [D:\a\realm-js\realm-js\packages\bindgen\build\TypeScript.vcxproj]
  If you are installing esbuild with npm, make sure that you don't specify the
  "--no-optional" or "--omit=optional" flags. The "optionalDependencies" feature
  of "package.json" is used by esbuild to install the correct binary executable
  for your current platform.
```

This was most likely related to the rebuild of the package-lock.json that it introduced.

These optional dependencies got added back in with https://github.com/realm/realm-js/pull/5652/files#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519

But on that commit Windows ia32 started failing with:

```

D:\a\realm-js\realm-js\vendor\realm-core\src\realm/array.hpp(569,16): warning C4146: unary minus operator applied to unsigned type, result still unsigned [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(695,24): error C2666: 'Napi::CallbackInfo::operator []': 2 overloads have similar conversions [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\node_modules\node-addon-api\napi-inl.h(3477,34): message : could be 'const Napi::Value Napi::CallbackInfo::operator [](size_t) const' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(695,24): message : or       'built-in C++ operator[(napi_callback_info, int)' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(695,24): message : while trying to match the argument list '(const Napi::CallbackInfo, int)' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(695,43): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(696,46): error C3536: 'ctors': cannot be used before it is initialized [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(696,79): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(697,77): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(698,85): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(699,89): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(700,91): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(701,99): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(702,83): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(703,81): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(704,95): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(705,85): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(706,75): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(707,91): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(708,91): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(709,87): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(710,83): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(711,97): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(712,79): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(713,97): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(714,87): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(715,83): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(716,79): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(717,101): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(718,97): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(719,103): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(720,85): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(721,89): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(722,77): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(723,75): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(724,89): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(725,97): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(726,95): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(727,97): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(729,84): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(730,85): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(731,91): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(732,109): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(733,75): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(734,91): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(735,89): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(737,89): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(739,83): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(740,95): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(741,91): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(742,107): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(743,95): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(744,91): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(745,107): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(747,85): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(748,87): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(749,115): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
D:\a\realm-js\realm-js\packages\bindgen\build\node_init.cpp(750,103): error C2059: syntax error: ')' [D:\a\realm-js\realm-js\packages\bindgen\build\realm-js.vcxproj]
....
```

The diff between the last green build and the first failing commit doesn't show any changes to the bindgen template: https://github.com/realm/realm-js/compare/e41501e5789c7d45ad5d3b580a911c7259ab1a55...18140b1eee4037fc84a4c523503c7737a2f0d728

--- a/package.json
+++ b/package.json
@@ -86,7 +86,7 @@
     "eslint-plugin-prettier": "^4.2.1",
     "eslint-plugin-react": "^7.31.10",
     "eslint-plugin-react-native": "^4.0.0",
-    "node-addon-api": "^5.0.0",
+    "node-addon-api": "^6.0.0",
     "react-native": "0.71.4",
     "rollup": "^3.15.0",
     "tsx": "^3.12.3",

